### PR TITLE
Support podman-remote kill container(s)

### PR DESCRIPTION
**Pull request by jwhonce**
Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
LGTM

File: cmd/podman/kill.go [hunk 1]
```diff
@@ -51,54 +50,45 @@ func init() {
...
-		maxWorkers = c.GlobalFlags.MaxWorks
+	for _, id := range ok {
+		fmt.Println(id)
 	}
-	logrus.Debugf("Setting maximum workers to %d", maxWorkers)
 
-	killErrors, errCount := shared.ParallelExecuteWorkerPool(maxWorkers, killFuncs)
-	return printParallelOutput(killErrors, errCount)
+	if len(failures) > 0 {
+		keys := reflect.ValueOf(failures).MapKeys()
```
Comment (mheon) [inline cmd/podman/kill.go]: Oh dear... Are we using this because the type differs between remote and local?

File: pkg/adapter/containers.go [hunk 2]
```diff
@@ -70,3 +72,32 @@ func (r *LocalRuntime) StopContainers(ctx context.Context, cli *cliconfig.StopVa
...
+
+// KillContainers sends signal to container(s) based on CLI inputs.
+// Returns list of successful id(s), map of failed id(s) + error, or error not from container
+func (r *LocalRuntime) KillContainers(ctx context.Context, cli *cliconfig.KillValues) ([]string, map[string]error, error) {
+	var (
+		ok       = []string{}
+		failures = map[string]error{}
+	)
+
+	killSignal := uint(syscall.SIGTERM)
```
Comment (mheon) [inline pkg/adapter/containers.go]: This is duped between local and remote - can we move this out into cmd/podman and share it between them?

Comment (jwhonce) [reply pkg/adapter/containers.go]: Currently the varlink and container API use different int types.

/cc baude

File: cmd/podman/kill.go (hunk 1 above)
Comment (jwhonce) [reply cmd/podman/kill.go]: Building a slice containing the map's keys to support same output as previous code.

File: pkg/adapter/containers.go (hunk 2 above)
Comment (mheon) [reply pkg/adapter/containers.go]: Oh. Oh dear.

File: cmd/podman/kill.go [hunk 3]
```diff
@@ -51,54 +50,45 @@ func init() {
...
 	}
-	defer runtime.Shutdown(false)
 
-	if c.Signal != "" {
+	if c.Flags().Changed("signal") {
 		// Check if the signalString provided by the user is valid
 		// Invalid signals will return err
-		sysSignal, err := signal.ParseSignal(c.Signal)
-		if err != nil {
+		if _, err := signal.ParseSignal(c.Signal); err != nil {
```
Comment (rhatdan) [inline cmd/podman/kill.go]: Why not do this always.  c.Signal defaults to KILL, so it should work nad no need to check the Changed flag above.

File: pkg/adapter/containers.go (hunk 2 above)
Comment (rhatdan) [reply pkg/adapter/containers.go]: Lets make them  use the same.

File: pkg/adapter/containers.go [hunk 4]
```diff
@@ -70,3 +72,32 @@ func (r *LocalRuntime) StopContainers(ctx context.Context, cli *cliconfig.StopVa
...
+	if cli.Flags().Changed("signal") {
+		sysSignal, _ := signal.ParseSignal(cli.Signal)
```
Comment (rhatdan) [inline pkg/adapter/containers.go]: Why isn't the error checked here?

File: pkg/adapter/containers_remote.go [hunk 5]
```diff
@@ -148,6 +150,36 @@ func (r *LocalRuntime) StopContainers(ctx context.Context, cli *cliconfig.StopVa
...
+// Returns list of successful id(s), map of failed id(s) + error, or error not from container
+func (r *LocalRuntime) KillContainers(ctx context.Context, cli *cliconfig.KillValues) ([]string, map[string]error, error) {
+	var (
+		ok       = []string{}
+		failures = map[string]error{}
+	)
+
+	killSignal := int64(syscall.SIGTERM)
+	if cli.Flags().Changed("signal") {
+		sysSignal, _ := signal.ParseSignal(cli.Signal)
```
Comment (rhatdan) [inline pkg/adapter/containers_remote.go]: Why aren't we checking failure?

No need to check if Signal .Changed, always parse.

File: pkg/adapter/containers.go (hunk 4 above)
Comment (jwhonce) [reply pkg/adapter/containers.go]: Error already checked upstream.  Can add it here

File: pkg/adapter/containers_remote.go (hunk 5 above)
Comment (jwhonce) [reply pkg/adapter/containers_remote.go]: Error already checked upstream.  Can add it here

File: pkg/adapter/containers.go (hunk 2 above)
Comment (jwhonce) [reply pkg/adapter/containers.go]: varlink currently doesn't support unsigned types while the container API uses an uint for the signal interface.  I pushed the difference down into the remote/local code.  Let me know if you like this better.

File: cmd/podman/kill.go (hunk 3 above)
Comment (jwhonce) [reply cmd/podman/kill.go]: done

**Comment by rhatdan** [comment]
@cevich Can we get failed tests to run multiple times on failures.  Seems like Humans are rerunning the same tests over and over again.  Especially the Ubuntu and RHEL7 tests.

**Comment by cevich** [comment]
@rhatdan ginkgo does this (to a degree) for free (the `-flateattempts 3` option).  I get what you're saying though.  RHEL should go away when RHEL testing goes away.  The Ubuntu "pod on pod top" (or something) is a real (honest) failure of some sort.  Re-running it automatically won't fix anything, it'll just make people wait longer for results that accidentally passed :frowning_face: (hiding the problem)

Chatted with @baude and @mheon about this on Monday night.  I'm afraid that some actual de-buggering is called for in this instance, or it will bite us harder later down the road :disappointed:  We don't think it's a flake b/c of an external dependency (or it would fail outside of Ubuntu) :confused: Brent's guess is something relating to AppArmor...which seems plausible but needs checking.

**Pull request merged**